lib/utils/find-valid-install.js

```diff
@@ -0,0 +1,38 @@
+const {dirname} = require('path');
+const debug = require('debug')('ghost-cli:find-instance');
+const chalk = require('chalk');
+const checkValidInstall = require('./check-valid-install');
+
+const isRoot = () => dirname(process.cwd()) === process.cwd();
+
+const die = (name) => {
+    console.error(`${chalk.yellow('Working directory is not a recognisable Ghost installation.')}
+Run \`ghost ${name}\` again within a folder where Ghost was installed with Ghost-CLI.`);
+    process.exit(1);
+};
+
+function findValidInstallation(name = '', recursive = false) {
+    while (!isRoot()) {
+        debug(`Checking valid install: ${process.cwd()}`);
+        const isValidInstall = checkValidInstall(name);
+
+        if (isValidInstall) {
+            return;
+        }
+
+        if (isValidInstall === false) {
+            process.exit(1);
+        }
+
+        if (!recursive) {
+            return die(name);
+        }
+
+        debug(`Going up...`);
+        process.chdir('..');
```

Comment:
apologies if I mentioned this in a comment before 😅 but would it make sense to, instead of `chdir`ing each time, to maintain `dir` string throughout the while loop, and just update it for each iteration where it doesn't find a directory?

i.e.
```js
let dir = process.cwd();
while(!isRoot(dir)) {
  const isValidInstall = checkValidInstall(name, dir);

  // checks
  debug('Going up...');
  dir = dirname(dir);
}
```
then do a chdir at the end? might be more performant

Reply:
Taking a look